use-cases/src/Plutus.hs

```diff
@@ -0,0 +1,138 @@
+{-# LANGUAGE EmptyDataDecls             #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+-- This is a mock of (parts of) the Plutus API
+module Plutus (
+
+  Address, PubKey, KeyPair, Value, standardTxFee, pubKeyAddress,
+
+  T, Tx(..), TxOut(..), TxOutRef(..), Witness, PlutusTx, Validator,
+  submitTransaction, assert, lookupMyPubKey, lookupMyKeyPair, createPayment, txInSign,
+  Range(..),
+  EventTrigger(..)
+) where
+
+import           Language.Haskell.TH        hiding (Range)
+import           Language.Haskell.TH.Syntax hiding (Range)
+{-# ANN module "HLint: ignore Avoid restricted function" #-}
+
+-- |Cardano address
+--
+type Address = String
+
+-- |Ada value
+--
+newtype Value = Value Integer
+              deriving (Eq, Ord, Num, Show, Read)
+
+-- |Public key
+--
+data PubKey
+
+instance Lift PubKey where
+  lift = undefined
+
+-- |Public key pair (no lift instance, because we never ought to put it into a transaction)
+--
+data KeyPair
+
+pubKeyAddress :: KeyPair -> Address
+pubKeyAddress = undefined
+
+data TxState
+
+-- |Transaction monad for coordination layer computations; provides access to the blockchain
+--
+newtype T a = T (TxState -> (TxState, a))
+
+instance Functor T where
+  fmap  = undefined
+
+instance Applicative T where
+  pure  = undefined
+  (<*>) = undefined
+
+instance Monad T where
+  return = undefined
+  (>>=)  = undefined
+
+-- Submit the given transaction to the blockchain
+--
+submitTransaction :: Tx -> T TxOutRef
+submitTransaction = undefined
+
+assert :: Bool -> T ()
+assert = undefined
+
+-- part of the wallet interface
+lookupMyPubKey :: T PubKey

```

Comment:
shouldn't this just be a function `KeyPair -> PubKey`, and then you already have a function to get the first?